Docs: Fix Build & HTML5

ax3l opened the pull request:
Fix RTD build and HTML5 issues in rendering.

Add a few Sphinx warnings (duplicate labels, ill-formed commands, outdated configs, etc.) on the way.

ax3l committed:
Docs: Fix Build & HTML5

Fix RTD build and HTML5 issues in rendering.

Add a few Sphinx warnings
(duplicate labels, ill-formed commands, outdated configs, etc.)
on the way.

ax3l commented on docs/source/install/profile.rst:
```diff
@@ -132,11 +132,11 @@ Queue: k80 (8x NVIDIA K80 12GB)
 -------------
 
-**System overview:** `link <https://www.olcf.ornl.gov/olcf-resources/compute-systems/summit/>`_
+**System overview:** `link <https://www.olcf.ornl.gov/olcf-resources/compute-systems/summit/>`__
```
Double underscores for links in RST create unnamed, in-place references. That avoids, that you get 10s of "the reference named `link` was already defined" warnings.

ax3l commented on docs/source/_static/hide_chapter_titles.css:
```diff
@@ -1,24 +1,24 @@
  */
-#how-to-read-this-document.section div.section {
-    display:none;
+section#how-to-read-this-document div.section {
```
This is the HTML5 update from newer sphinx and docutils versions.

ax3l commented on docs/source/testing/testbuilding.rst:
```diff
@@ -1,7 +1,7 @@
-.. _testing-usage:
+.. _testing-new:
```
Duplicate anchor label used elsewhere already. Renamed here.

BrianMarre commented:
@psychocoderHPC CI crashed with, could not connect to docker container

BrianMarre reviewed (commented):
looks good to me, an you actually read the description of my example ;)

psychocoderHPC commented:
@ax3l Thanks for fixing these things :heart: :heart: 

I restarted the failing tests and set the label that the CI is not required to run because these changes do not affect the code.

psychocoderHPC commented:
@ax3l Could you point to a source from where you aggregated all the knowledge about these changes in Sphinx?

ax3l commented:
I am humbled by the question because I fixed this just by looking at the build result, the build warnings and using my experience. That's why I added a few inline comments to the PR.

Otherwise, I usually look at the doc-linked rst primer and sphinx docs for introductories and detail questions.

@BrianMarre sure! Actually, when you compile sphinx locally there are a few more warnings visible about your parts of the docs, if you have the time to check them out.

BrianMarre commented:
is on my to do list after the version 2 pull request of FLYonPIC
